src/App.tsx

```diff
@@ -1,12 +1,62 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import GoodsList from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods?: Good[],
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  handleButtonClick = async (callback: Promise<Good[]>) => {
```

Comment:
Main purpose of this function is to deliver goods.
Also, better to pass exactly callback and not it's return value.
If you pass return values - there will be three API calls.
```suggestion
  getGoods = async (callback: () => Promise<Good[]>) => {
```